### DISCUSSION TITLE
Make Remote Buttons Accessible On Stream Deck

### DISCUSSION THREAD

**Discussion by illegalcereal** [PRProd/HA-Firemote#454] [Q&A] [answered]
### Feature Request

I absolutely love your product and use it all over the house on either my phone or tablet.  In my office, however, I try to use my Elgato Stream Deck for all of my HomeAssistant commands.  I am trying desperately to control my Chromecast 4k and Nvidia Shield TV Pro 2019 from the Stream Deck.  Such as Power, Up, Down, Left, Right, Select, Home, Options Button, etc.  Would it be possible to help narrow down the section of your code that controls the buttons for the Shield TV Pro 2019 and Chromecast 4k so that I can make a script to apply to each button on the Elgato Stream Deck?  Or possibly even the other devices you have available for your card?  I feel like adding this ability would draw even more people to your product.  While I am good at scripting, I don't necessarily know the language HA uses enough to figure this out on my own.
![Screenshot 2024-06-26 221558](https://github.com/PRProd/HA-Firemote/assets/107562422/702ffba8-437d-4193-85dc-6be6447352bc)
![Screenshot 2024-06-26 221712](https://github.com/PRProd/HA-Firemote/assets/107562422/89968151-defd-4c90-bbd2-e78bc4cc1308)
I have experience doing QA for software companies and have no qualms about testing anything you guys put together.  I hope you'll consider this.  Thank you.

**Comment by PRProd** [comment, marked as answer]
Greetings @illegalcereal!

Thank you for the follow and for your kind words, it truly means a lot to me!  Firemote is a passion project of mine, so I really love hearing things like this from people like you who appreciate my efforts.

I'm more than happy to help you sort out how to make these things work.  Before we begin, I first need to explain that Firemote does not directly communicate with any of the media player devices.  Instead, it's simply a button surface that sends commands through Home Assistant to existing integrations.

Take a look at the Firemote Wiki section titled [How Does Firemote Send commands to my devices?](https://github.com/PRProd/HA-Firemote/wiki#how-does-firemote-send-commands-to-my-devices) for this helpful diagram and more info.
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/f1564498-b338-4194-9245-95738484695a)

I don't have a Stream Deck, so I'm not confident enough to suggest the best path to try to accomplish your goal.  From the images you've posted though, it looks like you might only need a list of Home Assistant commands to send directly to Home Assistant Entites.  If so, you'll likely find a list like this intriguing/helpful to get started.  (taken from the latest [3.4.9](https://github.com/PRProd/HA-Firemote/releases/tag/v3.4.9) Firemote version)
 - [all buttons](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L4872-L5664)
 - [power button](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L4872-L4911)
 - [left](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L4965-L4983)
 - [right](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L5013-L5031)
 - [up](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L4945-L4963)
 - [down](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L5033-L5051)
 - [center / select](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L4985-L5011)
 - [home](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L5108-L5125)
 - [hamburger / options](https://github.com/PRProd/HA-Firemote/blob/v3.4.9/dist/HA-Firemote.js#L5127-L5146)

There's a lot of overlap between device types where a command for one could be the same command for the next.  So, maybe even reviewing the charts that I created for a few of the media players in an attempt to document all of the commands could be useful for you too:
 - [ADB Commands for onn. Google TV 4K Streaming Box (2023)](https://github.com/PRProd/HA-Firemote/wiki/onn.-Google-TV-4K-Streaming-Box-(2023)#adb-commands)
 - [ADB Commands for Fire TV Stick 4K 2nd Gen (2023)](https://github.com/PRProd/HA-Firemote/wiki/Fire-TV-Stick-4K-2nd-Gen-(2023)#adb-commands)
 - [ADB commands for Fire TV Stick 4K Max 2nd Gen (2023)](https://github.com/PRProd/HA-Firemote/wiki/Fire-TV-Stick-4K-Max-2nd-Gen-(2023)#adb-commands)
 - [ADB commands for Smart TV Fire TV 4 Series (2021)](https://github.com/PRProd/HA-Firemote/wiki/Smart-TV---Fire-TV-4-Series-(2021)#adb-commands)

With all of that in mind, I'm not going to close this issue with a "It doesn't work like that" comment, but instead I'm going to convert it into a discussion so we can work on it until you are satisfied.

I'm excited to help you get this working!  It might even inspire me to get a stream deck too.

-Doug

**Reply by illegalcereal** [reply]
I appreciate you even entertaining the idea of helping.  These look like a solid place to start for configuring the Stream Deck with the code you have put together.  I'm not in front of my Stream Deck right now, but I will certainly look at the documentation and see what I can implement.  I love how thorough you were in your response for something that is slightly-off the actual product itself.  You guys are great!

**Reply by PRProd** [reply]
I aim to please.  😄

**Reply by PRProd** [reply]
@illegalcereal ~

I'm curious if you were able to get something working.  I sure hope so!

**Reply by PRProd** [reply]
@illegalcereal ~ Could you do me a favor?  If the answer I gave above satisfied your question, could you please click the "Mark as answer" button, just so I know?  If not, I'm still interested in helping you on this.  Please let me know either way.  Thanks!

-Doug
